File: strategy/exercise/NotificationService.java
package strategy.exercise;

public class NotificationService implements NotificationSendingService {

    private final NotificationSendingService notificationSendingService;

    public NotificationService(NotificationSendingService notificationSendingService) {
        this.notificationSendingService = notificationSendingService;
    }

    public void sendNotification(User user, String message) {
        notificationSendingService.sendNotification(user, message);
    }
}

File: typescript/patterns/strategy/exercise/notification-service.ts
import type { NotificationSendingService } from "./notification-sending-service";
import type { User } from "./user";

export class NotificationService {
  constructor(
    private readonly notificationSendingService: NotificationSendingService
  ) {}

  sendNotification(user: User, message: string): void {
    this.notificationSendingService.sendNotification(user, message);
  }
}
